### PR TITLE
[flutter_tools] inject output preferences at the top level

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

The tool was setting the output preferences in a sub-context. Originally these were not injected before the arg parsers were created, though that was fixed by the lazy command creation. Once local engine is removed, the inner flutter_command Zone can be removed.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
This will also work towards unblocking testWithoutContext when  using TestFlutterCommandRunner

File: packages/flutter_tools/test/general.shard/output_preferences_test.dart [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    final OutputPreferences preferences = OutputPreferences.fromArguments(
+      <String>['--no-wrap'],
+      userMessages: userMessages,
+      stdio: stdio,
+    );
+
+    expect(preferences.showColor, true);
+    expect(preferences.wrapText, false);
+    // this calls context
+    // expect(preferences.wrapColumn, 80);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/output_preferences_test.dart:77]: is this supposed to be commented out?

File: packages/flutter_tools/lib/src/base/terminal.dart [hunk 2]
```diff
@@ -47,6 +50,41 @@ class OutputPreferences {
...
+      final List<String> parts = rawWrapColumn.split('=');
+      if (parts.length > 1) {
+        wrapColumn = int.tryParse(parts[1]);
+      }
+      if (wrapColumn == null) {
+        throwToolExit(userMessages.runnerWrapColumnInvalid(rawWrapColumn));
+      }
+    }
+
+    // If we're not writing to a terminal with a defined width, then don't wrap
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/base/terminal.dart:74]: Looks like the logic changed and this comment is no longer accurate.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**